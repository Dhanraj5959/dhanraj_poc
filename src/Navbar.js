import React from "react";
import { Link, Navigate } from "react-router-dom";

const Navbar = () => {
  return (
    <>
      <nav class="navbar navbar-expand-lg navbar-light">
        <Link class="navbar-brand" to="/">
          <h1 className="navhead" style={{ marginLeft: "100px" }}>
            {" "}
            <b className="title text-light" style={{ fontWeight: "900" }}>
              D-Bank
            </b>
          </h1>
        </Link>
        <button
          class="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarNavAltMarkup"
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse" id="navbarNavAltMarkup">
          <div class="navbar-nav" style={{ marginLeft: "900px" }}>
            {/* <li className="nav-item">
              <Link className="btn btn-dark" to="/userlogin">
                Login
              </Link>
            </li>
            <li className="nav-item  " style={{ marginLeft: "20px" }}>
              <Link className="btn btn-dark" to="/userregister">
                Register
              </Link>
            </li> */}
            {window.localStorage.getItem("app_token") ? (
              <>
                <li className="nav-item">
                  <Link className="btn btn-dark m-2" to="/deposit">
                    Deposit
                  </Link>
                </li>
                <li className="nav-item">
                  <Link className="btn btn-dark m-2" to="/banking">
                    Withdraw
                  </Link>
                </li>
                <li className="nav-item">
                  <Link
                    className="btn btn-dark m-2"
                    to="/userlogin"
                    onClick={() => {
                      window.localStorage.removeItem("app_token"); //removing the jwt token for logging out
                      Navigate(`/userlogin`);
                    }}
                  >
                    Logout
                  </Link>
                </li>
              </>
            ) : (
              <>
                <li className="nav-item">
                  <Link className="btn btn-dark" to="/userlogin">
                    Login
                  </Link>
                </li>
                <li className="nav-item  " style={{ marginLeft: "20px" }}>
                  <Link className="btn btn-dark" to="/userregister">
                    Register
                  </Link>
                </li>
              </>
            )}
          </div>
        </div>
      </nav>
    </>
  );
};

export default Navbar;
