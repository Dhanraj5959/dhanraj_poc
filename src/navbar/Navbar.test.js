import { render, screen } from "@testing-library/react";

import { renderComponent } from "../utils/test_helpers";
import Navbar from "./Navbar";

describe("Navbar page", () => {
  test("Should Render Intial Page", () => {
    render(renderComponent(<Navbar />));
    const text1 = screen.getAllByText(/D-Bank/i);
  });

  test("should test page link ", () => {
    render(renderComponent(<Navbar />));
    expect(screen.getByRole("link", { name: "Login" })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: "Register" })).toBeInTheDocument();
    // expect(screen.getByRole("link", { name: "Withdraw" })).toBeInTheDocument();
  });
  test("Matching the snapshot of the navbar", async () => {
    const { asFragment } = render(renderComponent(<Navbar />));
    expect(asFragment()).toMatchSnapshot();
  });
});
