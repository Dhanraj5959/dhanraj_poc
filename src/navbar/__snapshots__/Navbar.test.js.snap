// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page Matching the snapshot of the navbar 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg navbar-light"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      <h1
        class="navhead"
        style="margin-left: 100px;"
      >
         
        <b
          class="title text-light"
          style="font-weight: 900;"
        >
          D-Bank
        </b>
      </h1>
    </a>
    <button
      aria-controls="navbarNavAltMarkup"
      aria-expanded="false"
      aria-label="Toggle navigation"
      class="navbar-toggler"
      data-target="#navbarNavAltMarkup"
      data-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        class="navbar-nav"
        style="margin-left: 900px;"
      >
        <li
          class="nav-item"
        >
          <a
            class="btn btn-dark"
            href="/userlogin"
          >
            Login
          </a>
        </li>
        <li
          class="nav-item  "
          style="margin-left: 20px;"
        >
          <a
            class="btn btn-dark"
            href="/userregister"
          >
            Register
          </a>
        </li>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
